src/watchman/api_v1_handler.clj

```diff
@@ -65,7 +66,13 @@
       (do
         (models/remove-host-from-role (:id host) (:id role))
         {:status 204}) ; No content
-      (create-json-error-response 404 "hostname not found in role"))))
+      (create-json-error-response 404 "hostname not found in role")))
+
+  (POST "/snooze" {:keys [params role]}
```

Comment:
Instead of having to post a JSON body with a single ID (`{"id": 123}`), what if the API endpoint was `/roles/:role-id/snooze`? Then you could just pass the duration as a single query param.